components/TaskCard.vue

```diff
@@ -70,23 +93,85 @@ export default {
 
   created() {
     this.task = this.cardData;
+    this.taskDescription = this.task.description;
   },
   methods: {
     async markDone() {
-      this.loading = true;
-      await this.$store.dispatch('todos/changeTaskState', this.task);
+      if (!this.showEditIcon) {
+        if (this.taskDescription.length <= 0) {
+          this.titleInputError = true;
```

Comment:
use early return instead of nested ifs if possible.

Reply:
Resolved